ArrCharOps.java

```diff
@@ -126,7 +180,22 @@ public static long hashCode(char[] arr) {
      *         return -2 if there is an error with the input.
      */
     public static int compareTo(String str1, String str2) {
-        // Replace the following statement with your code
-        return 0;
-    }
+    if(str1 == "" || str2 == "") return -2;
+      
+        for (int i = 0; i < str1.length() && i < str2.length(); i++) {
+            if(str1.charAt(i) < str2.charAt(i)){
+                return -1 ;
+            } else if (str1.charAt(i) > str2.charAt(i)) {
+                return 1 ;
+            }
+        }
+        if (str1.length() < str2.length()){
+            return -1 ;
+        } else if (str1.length() > str2.length()) {
+            return 1 ;
+        } else if (str1.length() == str2.length()){
+            return 0;
+        }
```

Comment:
1. first 
`if (str1.length() < str2.length())` -> check if first is smaller in length (good)
`else if (str1.length() > str2.length())` -> check if second is smaller in length (good)
`else if (str1.length() == str2.length())` -> check if they are equals (not good)
if not entered the first 2 conditions it means that the first isn't smaller in length, and the second isn't smaller in length. 
In other words. all strings that are equal in length -> therefore `else` will be better

2. consider the following
```
if (str1.length() != str2.length()){
      return str1.length() > str2.length() ? 1 : -1;
}
return 0; 
```